x/inflation/genesis.go

```diff
@@ -13,18 +15,50 @@ func InitGenesis(
 	ak types.AccountKeeper,
 	data types.GenesisState,
 ) {
-	data.Minter.EpochProvisions = data.Params.GenesisEpochProvisions
-	k.SetMinter(ctx, data.Minter)
 	k.SetParams(ctx, data.Params)
-	ak.GetModuleAccount(ctx, types.ModuleName)
-	k.SetLastHalvenEpochNum(ctx, data.HalvenStartedEpoch)
+
+	// Ensure inflation module account is set on genesis
+	if acc := ak.GetModuleAccount(ctx, types.ModuleName); acc == nil {
+		panic("the inflation module account has not been set")
+	}
+
+	// Ensure unvested team module account is set on genesis
+	if acc := ak.GetModuleAccount(ctx, types.UnvestedTeamAccount); acc == nil {
+		panic("the unvested team module account has not been set")
+	}
+
+	// ensure team account exists
+	if data.Params.TeamAddress != "" {
+		teamAddr, err := sdk.AccAddressFromBech32(data.Params.TeamAddress)
+		if err != nil {
+			panic(fmt.Errorf("invalid team bech32 address: %w", err))
+		}
+
+		if acc := ak.GetAccount(ctx, teamAddr); acc == nil {
+			panic(fmt.Errorf("the team account %s has not been set", data.Params.TeamAddress))
+		}
+	}
+
+	// Set Period
+	period := data.Period
+	k.SetPeriod(ctx, period)
+
+	// Calculate epoch mint provision
+	epochMintProvision := types.CalculateEpochMintProvision(data.Params, period)
+	k.SetEpochMintProvision(ctx, epochMintProvision)
+
+	// Mint genesis coins for unvested team account
+	amount := sdk.NewInt(200_000_000)
+	coins := sdk.NewCoins(sdk.NewCoin(data.Params.MintDenom, amount))
+	if err := k.MintGenesisTeamVestingCoins(ctx, coins); err != nil {
+		panic(err)
+	}
```

Comment:
this will mint every time the chain is initialized (including upgrades). Can we add a bool or check if the team coins were already minted?